processor/batchprocessor/batch_processor.go

```diff
@@ -141,6 +143,16 @@ func (bp *batchTraceProcessor) startProcessingCycle() {
 	for {
 		select {
 		case td := <-bp.newTraceItem:
+			if bp.enforceBatchSize {
```

Comment:
What about some simpler approach. Add items as normal to the batch, then do a simple split not by max size. It is a bit more work maybe but simpler logic I feel.

Reply:
That might be even better from the perf standpoint. The buckets of max size would have to be probably split also bc there can be other items arriving in the meantime (unless we push them directly).